eth2/operation_pool/src/lib.rs

```diff
@@ -0,0 +1,952 @@
+use int_to_bytes::int_to_bytes8;
+use itertools::Itertools;
+use parking_lot::RwLock;
+use ssz::ssz_encode;
+use state_processing::per_block_processing::errors::{
+    AttestationValidationError, AttesterSlashingValidationError, DepositValidationError,
+    ExitValidationError, ProposerSlashingValidationError, TransferValidationError,
+};
+use state_processing::per_block_processing::{
+    gather_attester_slashing_indices_modular, validate_attestation,
+    validate_attestation_time_independent_only, verify_attester_slashing, verify_deposit,
+    verify_exit, verify_exit_time_independent_only, verify_proposer_slashing, verify_transfer,
+    verify_transfer_time_independent_only,
+};
+use std::collections::{btree_map::Entry, hash_map, BTreeMap, HashMap, HashSet};
+use types::chain_spec::Domain;
+use types::{
+    Attestation, AttestationData, AttesterSlashing, BeaconState, ChainSpec, Deposit, Epoch,
+    ProposerSlashing, Transfer, Validator, VoluntaryExit,
+};
+
+#[cfg(test)]
+const VERIFY_DEPOSIT_PROOFS: bool = false;
+#[cfg(not(test))]
+const VERIFY_DEPOSIT_PROOFS: bool = false; // TODO: enable this
```

Comment:
@paulhauner Why was this disabled? Safe to re-enable?

Reply:
For records sake, we'll leave this disabled until we add deposit proof generation to `test_harness`